src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -34,4 +48,47 @@ public function __construct(ContainerBuilder $container, FileLocatorInterface $l
 
         parent::__construct($locator);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function import($resource, $type = null, $ignoreErrors = false, $sourceResource = null)
+    {
+        if (strlen($resource) === $i = strcspn($resource, '*?{[')) {
+            $directoryPrefix = $resource;
+            $directoryGlob = '';
+        } else {
+            $directoryPrefix = dirname(substr($resource, 0, 1 + $i));
+            $directoryGlob = substr($resource, strlen($directoryPrefix));
+        }
+
+        try {
+            $directoryPrefix = $this->locator->locate($directoryPrefix, $this->currentDir, true);
+            $directoryPrefix = realpath($directoryPrefix) ?: $directoryPrefix;
+            $directoryGlob = $directoryPrefix.$directoryGlob;
+
+            if (!$files = glob($directoryGlob, defined('GLOB_BRACE') ? GLOB_BRACE : 0)) {
+                throw new FileLoaderLoadException($directoryGlob, $sourceResource);
+            }
+
+            foreach ($files as $file) {
+                if (is_dir($file)) {
+                    $this->container->addResource(new DirectoryResource($file, '/^$/'));
+                    parent::import($file, 'directory', $ignoreErrors, $sourceResource);
+                } else {
+                    $this->container->addResource(new FileResource($file));
```

Comment:
Now that we have #21408, resource tracking should be done using the new `$container->fileExists()` method, both here and above instead of Directory/FileResource.